msal-auth-node/README.md

```diff
@@ -0,0 +1,146 @@
+| **Scopes defined by this API** | **Scope name**: `Greeting.Read`<br/>**Who can consent?**: **Admins and users**<br/>**Admin consent display name**: `Read API Greetings`<br/>**Admin consent description**: `Allows the user to see greetings from the API.`<br/>**User consent display name**: `Read API Greetings`<br/>**User consent description**: `Allows you to see greetings from the API.`<br/>**State**: **Enabled** | Required scope for this sample. |
+
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+```javascript
+auth: {
+  // 'Directory (tenant) ID' of app registration in the Azure portal - this value is a GUID
+  tenant: '',
+
+  // 'Application (client) ID' of app registration in the Azure portal - this value is a GUID
+  audience: ''
+}
+```
+
+### 3. Install package(s)
+
+To install Node.js libraries:
+
+```bash
+npm install
+```
+
+## Run the application
+
+```bash
+node app.js
+```
+
+## Browse to the application
+
+Using Postman, curl, or a similar application, issue an HTTP GET request to *http://localhost:8080/me* with an `Authorization` header of `Bearer {VALID-ACCESS-TOKEN}`.
+
+For example, if you use curl and everything worked, the sample you should receive a response from the API similar to this:
+
+```console
+$ curl http://localhost:8080/me -H "Authorization: Bearer {VALID-ACCESS-TOKEN}"
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users/$entity",
+  "businessPhones": ["+1 (999) 5551001"],
+  "displayName": "Contoso Employee",
+  "givenName": "Contoso",
+  "jobTitle": "Worker",
+  "mail": "cemployee@contoso.com",
+  "mobilePhone": "1 999-555-1001",
+  "officeLocation": "Contoso Plaza/F30",
+  "preferredLanguage": null,
+  "surname": "Employee",
+  "userPrincipalName": "contoso_employee@contoso.com",
+  "id": "e3a49d8b-d849-48eb-9947-37c1f9589812"
+}
+```
+
+### Generating a valid access token
+
+Follow the instructions in [Gaining consent for the middle-tier application](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow#gaining-consent-for-the-middle-tier-application), including setting the API's app registration manifest value of **knownClientApplications**.
+
+## About the code
+
+This Node.js API has a single route that requires an access token. The access token will be automatically validated by MSAL. A missing or invalid (expired, wrong audience, etc) token will result in a `401` response. An otherwise valid token without the proper scopes will result in a `403` response. A valid token with a proper scope of (`Greeting.Read`) will result in a "Hello, world" message.
```

Comment:
You have the formatted version of this at the top of the file instead of down here.

Reply:
Thanks for moving it.